R/qenv-extract.R

```diff
@@ -0,0 +1,50 @@
+#'
+#' @section Subsetting:
+#' `x[names]` subsets objects in `qenv` environment and limit the code to the necessary needed to build limited objects.
+#' `...` passes parameters to further methods.
+#'
+#' @param x (`qenv`)
+#'
+#' @examples
+#'
+#' # Subsetting
+#' q <- qenv()
+#' q <- eval_code(q, "a <- 1;b<-2")
+#' q["a"]
+#' q[c("a", "b")]
+#'
+#' @rdname qenv
+#'
+#' @export
+`[.qenv` <- function(x, names, ...) {
```

Comment:
An `NA` artifact is created when subsetting the module's datanames with dot prefix such as `.raw_data`.

```r
pkgload::load_all("teal.code")
pkgload::load_all("teal.data")
pkgload::load_all(".")

app <- init(
  data = teal_data() |> within(iris <- iris),
  modules = example_module()
) |> shiny::runApp()
```

![image](https://github.com/user-attachments/assets/40f983a6-b6a7-4cda-bf99-95f7d94228a3)

![image](https://github.com/user-attachments/assets/3294700c-c562-4b87-8df7-e044eec8d42c)